### PR TITLE
feat: Remove duplicate information in the initiate negotiation api request

### LINKED ISSUE #3395: duplicate information in the initiate negotiation api request
# Feature Request

Currently in the `POST /v2/contractnegotiations` request the `ContractRequest` body has this structure:
```json
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "odrl": "http://www.w3.org/ns/odrl/2/"
  },
  "@type": "ContractRequest",
  "connectorId": "provider",
  "connectorAddress": "http://localhost:19194/protocol",
  "consumerId": "consumer",
  "providerId": "provider",
  "protocol": "dataspace-protocol-http",
  "offer": {
    "offerId": "MQ==:YXNzZXRJZA==:YTc4OGEwYjMtODRlZi00NWYwLTgwOWQtMGZjZTMwMGM3Y2Ey",
    "assetId": "assetId",
    "policy": {
      "@id": "MQ==:YXNzZXRJZA==:YTc4OGEwYjMtODRlZi00NWYwLTgwOWQtMGZjZTMwMGM3Y2Ey",
      "@type": "use",
      "odrl:permission": [],
      "odrl:prohibition": [],
      "odrl:obligation": [],
      "odrl:target": "assetId"
    }
  }
}
```

As you can see, the `offerId` and the `policy/@id` have the same value, because they're representing the same information, so there's no point in `offerId`, as the `policy` is mandatory anyway.

The proposal would be to remove `offerId`, and maybe inline the `offer` object and have `assetId` and `policy` at the root level, to have something like:
```json
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "odrl": "http://www.w3.org/ns/odrl/2/"
  },
  "@type": "ContractRequest",
  "connectorId": "provider",
  "connectorAddress": "http://localhost:19194/protocol",
  "consumerId": "consumer",
  "providerId": "provider",
  "protocol": "dataspace-protocol-http",
  "assetId": "assetId",
  "policy": {
    "@id": "MQ==:YXNzZXRJZA==:YTc4OGEwYjMtODRlZi00NWYwLTgwOWQtMGZjZTMwMGM3Y2Ey",
    "@type": "use",
    "odrl:permission": [],
    "odrl:prohibition": [],
    "odrl:obligation": [],
    "odrl:target": "assetId"
  }
}
```

## Which Areas Would Be Affected?

api

## Why Is the Feature Desired?

simplify

## Solution Proposal

this shouldn't be a breaking change, deprecation model should be used.

### PR DESCRIPTION
**Pull request by tuncaytunc-zf**
## What this PR changes/adds

Removes the duplicate information in the initiate negotiation api request without increasing the API version

## Why it does that

The offer and the policy have some duplicate values like id and assetId which are representing the same information.

## Further notes

The offer will be logged as deprecates to avoid any braking changes

## Linked Issue(s)

Closes #3395 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/validation/ContractRequestValidator.java [hunk 1]
```diff
@@ -16,28 +16,17 @@
...
     public static Validator<JsonObject> instance() {
         return JsonObjectValidator.newValidator()
                 .verify(CONNECTOR_ADDRESS, MandatoryValue::new)
                 .verify(PROTOCOL, MandatoryValue::new)
-                .verify(OFFER, MandatoryObject::new)
-                .verifyObject(OFFER, v -> v
-                        .verify(OFFER_ID, MandatoryValue::new)
-                        .verify(ASSET_ID, MandatoryValue::new)
-                        .verify(POLICY, MandatoryObject::new)
-                )
```
[the remark this answers is not in the document]
Comment (tuncaytunc-zf) [reply extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/validation/ContractRequestValidator.java:40]: I have extended the Validation but the logs are still in Transformator

File: extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/ContractNegotiationApi.java [hunk 2]
```diff
@@ -142,18 +142,14 @@ record ContractRequestSchema(
...
-                            }]
-                        }
+                    "policy": {
+                        "@context": "http://www.w3.org/ns/odrl.jsonld",
+                        "@type": "Set",
+                        "@id": "policy-id",
+                        "permission": [{
+                            "target": "asset-id",
+                            "action": "display"
+                        }]
```
[the remark this answers is not in the document]
Comment (tuncaytunc-zf) [reply extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/ContractNegotiationApi.java]: Fixed, thanks.

File: extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/ContractNegotiationApi.java [hunk 3]
```diff
@@ -131,7 +132,10 @@ record ContractRequestSchema(
             String connectorAddress,
             @Schema(requiredMode = REQUIRED)
             String providerId,
+            @Deprecated(since = "0.3.2")
+            @Schema(deprecated = true, description = "please use policy instead of offer")
             ContractOfferDescriptionSchema offer,
+            Policy policy,
```
Comment (ndr-brt) [inline extensions/control-plane/api/management-api/contract-negotiation-api/src/main/java/org/eclipse/edc/connector/api/management/contractnegotiation/ContractNegotiationApi.java:138]:
```suggestion
            ManagementApiSchema.PolicySchema policy,
```

**Review by ndr-brt — CHANGES_REQUESTED** [review]
last little nit, after that we'll be ready to merge

**Pull request merged**